python/dgl/convert.py

```diff
@@ -633,28 +655,38 @@ def to_homogeneous(G, ndata=None, edata=None):
     offset_per_ntype = np.insert(np.cumsum(num_nodes_per_ntype), 0, 0)
     srcs = []
     dsts = []
-    etype_ids = []
-    eids = []
-    ntype_ids = []
     nids = []
+    eids = []
+    if store_type:
+        ntype_ids = []
+        etype_ids = []
+    if return_count:
+        ntype_count = []
+        etype_count = []
```

Comment:
Can we not use a Python list?  We can create an `np.array` beforehand.

Reply:
The count array is usually used together with `torch.split` to compute messages one type at a time. `torch.split` requires python list: https://pytorch.org/docs/stable/generated/torch.split.html.